File: hjudge.Core/src/ExecOptions.cs
namespace hjudge.Core
{
    class ExecOptions
    {
        public string? Exec { get; set; }
        public string? Args { get; set; }
        public string? WorkingDir { get; set; }
        public string? StdErrRedirectFile { get; set; }
        public string? InputFile { get; set; }
        public string? OutputFile { get; set; }
        public long TimeLimit { get; set; }
        public long MemoryLimit { get; set; }
        public bool IsStdIO { get; set; }
        public int ActiveProcessLimit { get; set; }
    }
}

File: hjudge.Exec/src/exec_options.rs
use serde::{Deserialize, Serialize};
#[derive(Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct ExecOptions {
    pub exec: String,
    pub args: String,
    pub working_dir: String,
    pub std_err_redirect_file: String,
    pub input_file: String,
    pub output_file: String,
    pub time_limit: i64,
    pub memory_limit: i64,
    pub is_std_io: bool,
    pub active_process_limit: u32,
}
